index.js

```diff
@@ -0,0 +1,116 @@
+const { createInterface } = require('node:readline');
+const { ArgumentParser, BooleanOptionalAction } = require('argparse');
+const {fetchAMIs, deleteAMIs} = require('./lib.js');
+
+const AWS = require('aws-sdk');
+const PrettyTable = require('./prettytable.js');
+
+const rl = createInterface({
+  input: process.stdin,
+  output: process.stdout,
+}); 
+
+async function input(query) {
+  return new Promise(resolve => {
+    rl.question(query, answer => resolve(answer));
+  });
+}
+
+async function run({
+  includeName,
+  includeTagKey,
+  includeTagValue,
+  excludeNewest,
+  excludeInUse,
+  excludeDays,
+  forceDelete,
+  verbose
+}) {
+  const now = Date.now();
+  const ec2 = new AWS.EC2({apiVersion: '2016-11-15'});
+  const autoscaling = new AWS.AutoScaling({apiVersion: '2011-01-01'});
+
+  let amis = await fetchAMIs(now, ec2, autoscaling, includeName, includeTagKey, includeTagValue, excludeNewest, excludeInUse, excludeDays);
+
+  if (verbose === true) {
+    const pt = new PrettyTable();
+    pt.sortTable('Name');
+    pt.create(['ID', 'Name', 'Creation Date', 'Delete?', 'Include reasons', 'Exclude reasons'], amis.map(ami => [
+      ami.id,
+      ami.name,
+      new Date(ami.creationDate).toISOString(),
+      (ami.included === true && ami.excluded === false) ? 'yes' : 'no',
+      (ami.included === true) ? ami.includeReasons.join(', ') : '',
+      (ami.excluded === true) ? ami.excludeReasons.join(', ') : ''
+    ]));
+    pt.print();
+  }
+
+  amis = amis.filter(ami => ami.included === true && ami.excluded === false);
+
+  if (amis.length === 0) {
+    return;
+  }
+
+  let del = false;
+  if (forceDelete === true) {
+    del = true;
+  } else {
+    const answer = await input(`Do you want to continue and remove ${amis.length} AMIs [y/N] ? : `);
+    del = answer === 'y';
+  }
+
+  if (del === true) {
+    await deleteAMIs(ec2, amis);
+  } else {
+    return;
+  }
+}
+
+function readArgs() {
```

Comment:
How to configure the region?

Reply:
The only wait is to set the `AWS_REGION` environment variable at the moment. Should we add and arg?

Reply:
I'd say it would be worth the effort.

Reply:
@andreaswittig done